src/test/java/reposense/parser/ArgsParserTest.java

```diff
@@ -47,8 +47,33 @@
 
     @Test
     public void parse_allCorrectInputs_success() throws ParseException, IOException {
-        String input = String.format("-config %s -output %s -since 01/07/2017 -until 30/11/2017 "
-                + "-formats java adoc html css js --ignore-standalone-config -view",
+        String input = String.format("--config %s --output %s --since 01/07/2017 --until 30/11/2017 "
+                + "--formats java adoc html css js --ignore-standalone-config --view",
```

Comment:
Still using magic strings throughout the test though?

Reply:
I am not so sure if it is better to substitute all the string here. Because, if we do so, if there is a typo in the flags, we are unable to find it out by tests. Also, it is not so intuitive from my perspective.

Reply:
>Because, if we do so, if there is a typo in the flags, we are unable to find it out by tests.

If we use the variable as given in `ArgsParser` why would there be a typo when they are exactly the same? Rather, now since we are still using magic Strings here, if in the future we do change the option name in `ArgsParser` we would need to update it here as well (like what you are doing now, converting from single `-` to double `--`), which is inconvenient.

>Also, it is not so intuitive from my perspective.

How would it become not intuitive? We could separate the creation of each option and its argument, and then finally join them together using `String.join`, before running the parsing test, which I think would make it more readable as well.

@yong24s Your opinion on this?

Reply:
Clearly, like you say, using constants would be easier to maintain. Also, we are unlikely to make  typos since we have all the peer reviews. 👀

But we are unlikely to do such a major change to the flags again. 🤔 Also the only benefit of using magic string is its easier to read off the tests.

Either way are fine for me. Ask @eugenepeh too. 😬

Reply:
If that's the concern, let's make a test util which builds the command string.

Reply:
Because there are many combinations of args, can I know what kind of function do you expect to have in test Util? For example, input and output? 
I am considering the suggestion by yamidark:
```
How would it become not intuitive? We could separate the creation of each option and its argument, and then finally join them together using `String.join`, before running the parsing test, which I think would make it more readable as well.
```

Reply:
@fzdy1914 Can you try my suggestion 1st, maybe for the new tests 1st, then we can review and see if we want to change the others as well.

Reply:
>Because there are many combinations of args, can I know what kind of function do you expect to have in test Util? For example, input and output?

E.g. `new InputBuilder().addConfig("...").addSince("01/07/2017").build()` ... etc. Its basically a StringBuilder in the backend with concept similar to the builders in [AB4](https://github.com/se-edu/addressbook-level4/tree/master/src/test/java/seedu/address/testutil)

>@fzdy1914 Can you try my suggestion 1st, maybe for the new tests 1st, then we can review and see if we want to change the others as well.

We can do that too, whichever provides better readability.

Reply:
I have implemented the InputBuilder part, please kindly give a review and give suggestions about the alias method mentioned below.
Also, please kindly give a suggestion to improve the test below:
https://github.com/reposense/RepoSense/pull/526/files#diff-aca26ed6b596a7c94f290b9c15f01597R109

Reply:
since its a testutil, doesn't hurt to add a method such as `addWhiteSpace(int)` etc to help you with that test.

Reply:
@eugenepeh How about the alias builder?

Reply:
its fine to leave it as it is, no point fine graining it